courses/views_test.py

```diff
@@ -194,20 +195,28 @@ def test_course_catalog_view(client):
     """
     program = ProgramFactory.create(live=True)
 
-    course_in_program = CourseFactory.create(program=program, live=True)
-    course_no_program = CourseFactory.create(no_program=True, live=True)
-
     # Required because catalog now excludes Programs and Courses that do not have a page
-    ProgramPageFactory.create(program=program)
-    CoursePageFactory.create(course=course_in_program)
-    CoursePageFactory.create(course=course_no_program)
+    program_page = ProgramPageFactory.create(program=program)
+    course_page_in_program = CoursePageFactory.create(
+        course=CourseFactory.create(program=program, live=True)
+    )
+    course_page_no_program = CoursePageFactory.create(
+        course=CourseFactory.create(no_program=True, live=True)
+    )
     CourseFactory.create(no_program=True, live=False)
-    exp_programs = [program]
-    exp_courses = [course_in_program, course_no_program]
-    resp = client.get(reverse("mitxpro-catalog"))
-    assert resp.templates[0].name == "catalog.html"
-    assert list(resp.context["programs"]) == exp_programs
-    assert list(resp.context["courses"]) == exp_courses
+    exp_program_pages = [program_page]
+    exp_course_pages = [course_page_in_program, course_page_no_program]
+    catalog_page = CatalogPage.objects.first()
+    if not catalog_page:
+        catalog_page = CatalogPage(title="Catalog", slug="catalog")
+        catalog_page = Site.objects.get(is_default_site=True).root_page.add_child(
+            catalog_page, "last-child"
+        )
```

Comment:
Please create a fixture for (a) the root page (`Site.objects.get(is_default_site=True).root_page`), and (b) the catalog page (lines 210 to here). The catalog page fixture would user the root page fixture, then this test case would use the catalog page fixture. You can just define them at the top of this module

Reply:
Done